### PR TITLE
new pdf option noStdin

### PR DESCRIPTION
**Pull request by oleksabor**
In order to deal with #4488 I'd like to suggest to add new `noStdin` pdf option 
It can be set using `docfx.json` file to run `docfx pdf` on the Azure build pipeline.

````Json
"pdf": {
 "noStdin": true,    
"content": [

````

### CODE REVIEW

File: src/Microsoft.DocAsCode.HtmlToPdf/ConvertWrapper.cs [hunk 1]
```diff
@@ -328,15 +328,16 @@ private IList<HtmlModel> BuildHtmlModels(string basePath, IList<TocModel> tocMod
...
-            var converter = new HtmlToPdfConverter(
-                htmlModels,
-                new HtmlToPdfOptions
-                {
-                    BasePath = basePath,
-                    UserStyleSheet = _pdfOptions.CssFilePath,
-                    LoadErrorHandling = _pdfOptions.LoadErrorHandling,
-                    AdditionalArguments = _pdfOptions.AdditionalPdfCommandArgs,
-                    OutlineOption = _pdfOptions.OutlineOption
+			var converter = new HtmlToPdfConverter(
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.HtmlToPdf/ConvertWrapper.cs]: Replace <kbd>tab</kbd> with 4 <kbd>space</kbd>s

**Comment by superyyrrzz** [comment]
This PR makes me think whether do we need this option. Could we always disable `IsReadArgsFromStdin`?

/cc @partychen

File: src/Microsoft.DocAsCode.HtmlToPdf/ConvertWrapper.cs (hunk 1 above)
Comment (oleksabor) [reply src/Microsoft.DocAsCode.HtmlToPdf/ConvertWrapper.cs]: Please find tabs were replaced with spaces in the commit af6b07fe862fd572a7ee76b5607bd3e6bf273816

**Comment by superyyrrzz** [comment]
@oleksabor I just talked with @partychen . Looks like the root cause is stderr in wkhtmltopdf is not handled correctly. I will soon create another PR to fix this. We can first hold on this PR and see if that PR can work for you.

**Comment by oleksabor** [comment]
@superyyrrzz I see, thank you for information

I'm not in a hurry since have patched docfx using v2.42.4 for Azure pipelines. 

However I'd like to add that had tried to read from the `wkhtmltopdf`s `stderr` and failed - there is no output even if it returns 1
I've used `RedirectStandardError = config.RedirectError,` to redirect stderr and then read it as

````CSharp
if (IsOutputToStdout)
{
	Log.Debug("reading stdout");
	using (var standardOutput = process.StandardOutput)
	{
		//standardOutput.BaseStream.CopyTo(stream);
		using (var reader = new StreamReader(standardOutput.BaseStream))
		{
			bufStr = reader.ReadToEnd();
		}
	}
}
if (config.RedirectError)
{
	Log.Debug("reading stderr");
	using (var sout = process.StandardError)
	{
		using (var reader = new StreamReader(sout.BaseStream))
		{
			var error = reader.ReadToEnd();
			if (!string.IsNullOrEmpty(error))
				throw new ApplicationException(error);
		}
	}
}
var exited = process.WaitForExit(TimeoutInMilliseconds);
````

`var error = reader.ReadToEnd();` returns `null` even if `exited` is not zero and `wkhtmltopdf` does not output pdf.

I've tried to run `wkhtmltopdf` using Azure pipeline and it there is nothing in the `stderr` as well.

**Comment by superyyrrzz** [comment]
@oleksabor I created #4733. Could you help review and try whether it can fix your issue? From your latest comment, I feel this is what you have tried. I am not sure whether it can work.

**Comment by superyyrrzz** [comment]
Sorry for the late reply. I see this is a feasible fix for this issue, while there are still some changes required for this PR:
* Replace all tabs with 4 spaces
* "Update branch" to pick up latest changes in dev branch
* Update the table in `Documentation/tutorial/docfx.exe_user_manual.md` for this new option: https://dotnet.github.io/docfx/tutorial/docfx.exe_user_manual.html#33-properties-for-pdf

File: src/docfx/SubCommands/PdfCommand.cs [hunk 2]
```diff
@@ -184,10 +184,10 @@ private static void MergeOptionsToConfig(PdfCommandOptions options, PdfJsonConfi
                 config.GeneratesExternalLink = options.GeneratesExternalLink.Value;
             }
 
-			if (options.NoInputStreamArgs.HasValue)
-			{
-				config.NoInputStreamArgs = options.NoInputStreamArgs.Value;
-			}
+            if (options.NoInputStreamArgs.HasValue)
+            {
+	            config.NoInputStreamArgs = options.NoInputStreamArgs.Value;
```
Comment (superyyrrzz) [inline src/docfx/SubCommands/PdfCommand.cs]: This should be the last tab 😄

**Pull request merged**